seqr/views/apis/anvil_workspace_api_tests.py

```diff
@@ -375,44 +378,72 @@ def test_validate_anvil_vcf(self, mock_file_iter, mock_file_exist, mock_utils_lo
                                                          meta_fields=['workspace.bucketName'])
 
         # Test bad data path
-        mock_file_exist.return_value = False
+        mock_subprocess.return_value.wait.return_value = -1
+        mock_subprocess.return_value.stdout = [b'File not found']
         response = self.client.post(url, content_type='application/json', data=json.dumps(REQUEST_BODY_NO_SLASH_DATA_PATH))
         self.assertEqual(response.status_code, 400)
         self.assertEqual(response.json()['error'], 'Data file or path test_no_slash_path.vcf.bgz is not found.')
-        mock_file_exist.assert_called_with('gs://test_bucket/test_no_slash_path.vcf.bgz', user=self.manager_user)
+        mock_subprocess.assert_called_with('gsutil ls gs://test_bucket/test_no_slash_path.vcf.bgz', stdout=-1, stderr=-2, shell=True)
+        mock_file_logger.info.assert_has_calls([
+            mock.call('==> gsutil ls gs://test_bucket/test_no_slash_path.vcf.bgz', self.manager_user),
+            mock.call('File not found', self.manager_user),
+        ])
 
         response = self.client.post(url, content_type='application/json', data=json.dumps(REQUEST_BODY_BAD_DATA_PATH))
         self.assertEqual(response.status_code, 400)
         self.assertEqual(response.json()['error'],
                          'Invalid VCF file format - file path must end with .vcf or .vcf.gz or .vcf.bgz')
 
+        # Test bad sharded VCF file path
```

Comment:
does this actually test a different code path than the above test? It seems like both are just testing the case when the call to `ls` fails

Reply:
Yes, it tests a different code path. If the data path has a `*` in it, the code will use `get_gs_file_list` to get sharded VCFs. Otherwise, it calls `does_file_exist` to check file existence.